pkg/envoy/cds/builder.go

```diff
@@ -28,9 +31,139 @@ import (
 	"github.com/openservicemesh/osm/pkg/trafficpolicy"
 )
 
+type clusterBuilder struct {
+	proxyIdentity identity.ServiceIdentity
+
+	outboundMeshTrafficPolicy *trafficpolicy.OutboundMeshTrafficPolicy
+
+	inboundMeshTrafficPolicy *trafficpolicy.InboundMeshTrafficPolicy
+
+	sidecarSpec configv1alpha2.SidecarSpec
+
+	egressTrafficPolicy *trafficpolicy.EgressTrafficPolicy
+
+	envoyOutboundPassthroughCluster string
+
+	egressEnabled bool
+
+	metricsEnabled bool
+
+	tracingEnabled bool
+
+	tracingAddress string
+
+	tracingPort uint32
+}
+
+func NewClusterBuilder() *clusterBuilder {
+	return &clusterBuilder{}
+}
+
+func (b *clusterBuilder) SetProxyIdentity(identity identity.ServiceIdentity) *clusterBuilder {
+	b.proxyIdentity = identity
+	return b
+}
+
+func (b *clusterBuilder) SetOutboundMeshTrafficPolicy(outboundTrafficPolicy *trafficpolicy.OutboundMeshTrafficPolicy) *clusterBuilder {
+	b.outboundMeshTrafficPolicy = outboundTrafficPolicy
+	return b
+}
+
+func (b *clusterBuilder) SetSidecarSpec(spec configv1alpha2.SidecarSpec) *clusterBuilder {
+	b.sidecarSpec = spec
+	return b
+}
+
+func (b *clusterBuilder) SetProxyServices(proxyServices []service.MeshService) *clusterBuilder {
+	b.sidecarSpec = spec
+	return b
+}
+
+func (b *clusterBuilder) SetInboundMeshTrafficPolicy(inboundTrafficPolicy *trafficpolicy.InboundMeshTrafficPolicy) *clusterBuilder {
+	b.inboundMeshTrafficPolicy = inboundTrafficPolicy
+	return b
+}
+
+func (b *clusterBuilder) SetEgressTrafficPolicy(egressTrafficPolicy *trafficpolicy.EgressTrafficPolicy) *clusterBuilder {
+	b.egressTrafficPolicy = egressTrafficPolicy
+	return b
+}
+
+func (b *clusterBuilder) SetEnvoyOutBoundPassthroughCluster(envoyOutboundPassthroughCluster string) *clusterBuilder {
+	b.envoyOutboundPassthroughCluster = envoyOutboundPassthroughCluster
+	return b
+}
+
+func (b *clusterBuilder) SetEgressEnabled(egressEnabled bool) *clusterBuilder {
```

Comment:
nit: all of these exported methods will need comments to pass linting

Reply:
it seems lint doesn't seem to have an issue with this, but I can add them if you feel it's necessary